server/services/gameTick.ts

```diff
@@ -444,17 +607,15 @@ export default class GameTickService extends EventEmitter {
         return graph;
     }
 
-    _filterAvoidCarrierToCarrierCombatCarriers(carrierPositions: CarrierPosition[]): CarrierPosition[] {
-        return carrierPositions.filter(c => {
-            let specialist = this.specialistService.getByIdCarrier(c.carrier.specialistId);
-
-            if (specialist && specialist.modifiers && specialist.modifiers.special 
-                && specialist.modifiers.special.avoidCombatCarrierToCarrier) {
-                return false;
-            }
-
-            return true;
-        });
+    _checkAvoidCarrierToCarrierCombat(carrier) {
+        // Returns true if and only if the carrier avoids c2cc.
+        let specialist = this.specialistService.getByIdCarrier(carrier.specialistId);
+        return (
+            specialist
+            && specialist.modifiers
+            && specialist.modifiers.special
+            && specialist.modifiers.special.avoidCombatCarrierToCarrier
```

Comment:
We should use the special value functions in the `specialistService` to get this value.